crates/client/eth/src/l1_gas_price.rs

```diff
@@ -0,0 +1,276 @@
+use crate::client::EthereumClient;
+use alloy::eips::BlockNumberOrTag;
+use alloy::providers::Provider;
+use anyhow::Context;
+use dc_mempool::L1DataProvider;
+use futures::lock::Mutex;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::time::sleep;
+
+use lazy_static;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+lazy_static::lazy_static! {
+    static ref LAST_UPDATE_TIMESTAMP: Arc<Mutex<u128>> = Arc::new(Mutex::new(0));
+}
+
+// Function to update the last update timestamp
+pub async fn update_last_update_timestamp() {
+    let now = SystemTime::now().duration_since(UNIX_EPOCH).expect("Time went backwards").as_millis();
+    let mut timestamp = LAST_UPDATE_TIMESTAMP.lock().await;
+    *timestamp = now;
+}
+
+// Function to get the last update timestamp
+pub async fn get_last_update_timestamp() -> u128 {
+    *LAST_UPDATE_TIMESTAMP.lock().await
+}
+
+pub async fn gas_price_worker(
+    eth_client: &EthereumClient,
+    l1_data_provider: Arc<dyn L1DataProvider>,
+    infinite_loop: bool,
+    gas_price_poll_ms: u64,
+) -> anyhow::Result<()> {
+    update_last_update_timestamp().await;
+    loop {
+        match update_gas_price(eth_client, Arc::clone(&l1_data_provider)).await {
+            Ok(_) => log::trace!("Updated gas prices"),
+            Err(e) => log::error!("Failed to update gas prices: {:?}", e),
+        }
+
+        let last_update_timestamp = get_last_update_timestamp().await;
+        let current_timestamp = std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .expect("Failed to get current timestamp")
+            .as_millis();
+
+        if current_timestamp - last_update_timestamp > 10 * gas_price_poll_ms as u128 {
+            panic!(
```

Comment:
in case we are not able to fetch the latest gas prices, it was present in the madara code as well. I believe it exist to make sure we don't have gas fees deviation. but we can have a flag for this as well